wrapper/minecraft/src/main/kotlin/org/quiltmc/qkl/wrapper/minecraft/brigadier/BrigadierDsl.kt

```diff
@@ -0,0 +1,211 @@
+import com.mojang.brigadier.builder.RequiredArgumentBuilder
+import com.mojang.brigadier.context.CommandContext
+
+public typealias RequiredArgumentAction<S> = RequiredArgumentBuilder<S, *>.() -> Unit
+public typealias LiteralArgumentAction<S> = LiteralArgumentBuilder<S>.() -> Unit
+public typealias CommandAction<S> = CommandContext<S>.() -> Int
+
+/**
+ * Registers a command under [command] as the name.
+ *
+ * @author Oliver-makes-code (Emma)
+ */
+public fun <S> CommandDispatcher<S>.register(
+    command: String,
+    action: LiteralArgumentAction<S>
+) {
+    val argument = LiteralArgumentBuilder.literal<S>(command)
+    argument.apply(action)
+    register(argument)
+}
+
+/**
+ * Adds a literal with [name] as the literal.
+ *
+ * @author Oliver-makes-code (Emma)
+ */
+public fun <S, T : ArgumentBuilder<S, T>?> ArgumentBuilder<S, T>.literal(
+    name: String,
+    action: LiteralArgumentAction<S>
+) {
+    val argument = LiteralArgumentBuilder.literal<S>(name)
+    argument.apply(action)
+    then(argument)
+}
+
+/**
+ * Adds a string with [name] as the parameter name.
+ *
+ * @author Oliver-makes-code (Emma)
+ */
+public fun <S, T : ArgumentBuilder<S, T>?> ArgumentBuilder<S, T>.string(
+    name: String,
+    action: RequiredArgumentAction<S>
+) {
+    val argument = RequiredArgumentBuilder.argument<S, String>(
+        name,
+        StringArgumentType.string()
+    )
+    argument.apply(action)
+    then(argument)
+}
+
+/**
+ * Adds a greedy string with [name] as the parameter name.
+ *
+ * @author Oliver-makes-code (Emma)
+ */
+public fun <S, T : ArgumentBuilder<S, T>?> ArgumentBuilder<S, T>.greedyString(
+    name: String,
+    action: RequiredArgumentAction<S>
+) {
+    val argument = RequiredArgumentBuilder.argument<S, String>(
+        name,
+        StringArgumentType.greedyString()
+    )
+    argument.apply(action)
+    then(argument)
+}
+
+/**
+ * Adds a word with [name] as the parameter name.
+ *
+ * @author Oliver-makes-code (Emma)
+ */
+public fun <S, T : ArgumentBuilder<S, T>?> ArgumentBuilder<S, T>.word(
+    name: String,
+    action: RequiredArgumentBuilder<S, String>.() -> Unit = { }
+) {
+    val argument = RequiredArgumentBuilder.argument<S, String>(
+        name,
+        StringArgumentType.word()
+    )
+    argument.apply(action)
+    then(argument)
+}
+
+/**
+ * Adds a boolean with [name] as the parameter name.
+ *
+ * @author Oliver-makes-code (Emma)
+ */
+public fun <S, T : ArgumentBuilder<S, T>?> ArgumentBuilder<S, T>.boolean(
+    name: String,
+    action: RequiredArgumentAction<S>
+) {
+    val argument = RequiredArgumentBuilder.argument<S, Boolean>(
+        name,
+        BoolArgumentType.bool()
+    )
+    argument.apply(action)
+    then(argument)
+}
+
+/**
+ * Adds a double with [name] as the parameter name.
+ *
+ * @author Oliver-makes-code (Emma)
+ */
+public fun <S, T : ArgumentBuilder<S, T>?> ArgumentBuilder<S, T>.double(
+    name: String,
+    min: Double? = null,
```

Comment:
IMO these number parameters should just take a non-nullable value and we should just specify the default right in the method parameters, like this:
```kt
public fun <S, T : ArgumentBuilder<S, T>?> ArgumentBuilder<S, T>.double(
    name: String,
    min: Double = -Double.MAX_VALUE,
    max: Double = Double.MAX_VALUE,
    action: RequiredArgumentAction<S>
) {
```
Would make the default values a lot clearer, and on JVM we can just use `double`s instead of a reference type like `@Nullable Double`